mirascope/core/mistral/_utils/_convert_message_params.py

```diff
@@ -1,23 +1,35 @@
 """Utility for converting `BaseMessageParam` to `ChatMessage`."""
 
-from mistralai.models.chat_completion import ChatMessage
+from mistralai.models import (
+    AssistantMessage,
+    SystemMessage,
+    ToolMessage,
+    UserMessage,
+)
 
 from ...base import BaseMessageParam
 
 
 def convert_message_params(
-    message_params: list[BaseMessageParam | ChatMessage],
-) -> list[ChatMessage]:
+    message_params: list[
+        BaseMessageParam | AssistantMessage | SystemMessage | ToolMessage | UserMessage
+    ],
+) -> list[BaseMessageParam]:
     converted_message_params = []
     for message_param in message_params:
-        if isinstance(message_param, ChatMessage):
+        if not isinstance(
+            message_param,
+            BaseMessageParam,
+        ):
             converted_message_params.append(message_param)
         elif isinstance(content := message_param.content, str):
-            converted_message_params.append(ChatMessage(**message_param.model_dump()))
+            converted_message_params.append(
+                BaseMessageParam(**message_param.model_dump())
```

Comment:
this is not converted since by this point `message_param` is already confirmed to be a `BaseMessageParam` instance, so we'll need to check the role and convert this into the correct mistral-specific message param